File: src/server/Modules/Accounting/Modules.Accounting.Core/Entities/PayrollTransaction.cs
using FluentPOS.Shared.Core.Domain;
using FluentPOS.Shared.DTOs.Enums;
using System;

namespace FluentPOS.Modules.Accounting.Core.Entities
{
    public class PayrollTransaction : BaseEntity
    {
        public Guid PayrollId { get; set; }

        public TransactionType TransactionType { get; set; }

        public EntryType EntryType { get; set; }

        public string TransactionName { get; set; }

        public double Earning { get; set; }

        public double Deduction { get; set; }

        public double DaysOrHours { get; set; }

        public decimal PerDaySalary { get; set; }

    }
}

File: src/client/src/app/modules/admin/payroll/models/PayrollTransaction.ts
import { EntryType } from "src/app/core/enums/EntryType";
import { TransactionType } from "src/app/core/enums/TransactionType";

export interface PayrollTransaction {
    id: string;
    createaAt: string;
    updatedAt: string;
    organizationId: string;
    branchId: string;
    payrollRequestId: string;
    payrollId: string;

    transactionType: TransactionType;

    entryType: EntryType;

    transactionName: string;

    earning: number;

    deduction: number;

    daysOrHours: number;

    perDaySalary: number;
}
